.github/workflows/full-build-pr-and-main.yml

```diff
@@ -70,28 +88,70 @@ jobs:
         rm dist/Artemis_Grading_MacOS_aarch64.zip
         ditto -c -k --sequesterRsrc --keepParent Eclipse.app dist/Artemis_Grading_MacOS_aarch64.zip
 
+    - name: Save Eclipse Distribution (MacOS)
+      uses: actions/upload-artifact@v3
+      with:
+        name: "Eclipse Distribution MacOS"
+        path: "dist/Artemis_*_MacOS*.zip"
+
+  upload-windows:
+    name: "Upload of windows distributions"
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+
+    - name: Load Distributions from cache
+      uses: actions/cache@v3
+      with:
+        path: ./dist
+        key: cache-dist-${{github.run_id}}
+
     - name: Save Eclipse Distribution (Windows)
       uses: actions/upload-artifact@v3
       with:
         name: "Eclipse Distribution Windows"
         path: "dist/Artemis_*_Windows.zip"
-    
+
+  upload-linux:
+    name: "Upload of linux distributions"
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+
+    - name: Load Distributions from cache
+      uses: actions/cache@v3
+      with:
+        path: ./dist
+        key: cache-dist-${{github.run_id}}
+
+
     - name: Save Eclipse Distribution (Linux)
       uses: actions/upload-artifact@v3
       with:
         name: "Eclipse Distribution Linux"
         path: "dist/Artemis_*_Linux*.zip"
 
-    - name: Save Eclipse Distribution (MacOS)
-      uses: actions/upload-artifact@v3
+  release:
+    name: "Create release"
+    runs-on: ubuntu-latest
+    needs:
+      - upload-mac
+      - upload-windows
+      - upload-linux
+
+    steps:
+
+    - name: Load Distributions from cache
+      uses: actions/cache@v3
       with:
-        name: "Eclipse Distribution MacOS"
-        path: "dist/Artemis_*_MacOS*.zip"
-  
-          
+        path: ./dist
+        key: cache-dist-${{github.run_id}}
+
     - name: Release
       uses: softprops/action-gh-release@v1
-      if: startsWith(github.ref, 'refs/tags/v')
+        # if: startsWith(github.ref, 'refs/tags/v')
```

Comment:
I have to check if this part of the action is needed or not. I'd assume that this is just an old part that will not be executed anyone

Reply:
This change should be reverted. It is used if I trigger this workflow manually for a tagged version :)